### ISSUE TITLE
Add raw pointer access method to PoolVector

### ISSUE THREAD

**Issue by lupoDharkael** [godotengine/godot#24240]
I'm working with procedural generation in my project and I need to create hundreds of meshes and their respective normals, vertices, indices and for each level of detail.
I'd like to be able to use multithreading for the generation of the Array with the data of the mesh but the API requires the menipulation of PoolVector3Array and other types based on PoolVector in the engine so I'm forced to have a sequential generation of the meshes.

A method to have access to the raw pointer of the memory would be ideal. Maybe it could be called `get_ptr_unsafe` to make it clear it is not safe to use it for every usecase because the lack of thread safety and the posible invalidation of the pointer after a realloc.

In my case the mutex locking means a noticeably slower generation of the worlds and every mesh uses different PoolVectors so there is no danger of concurrent access.

The new method doesn't make sense in GDScript because you already sacrifice performance using an interpreted language, this is more oriented to GDNative languages.

**Comment by lupoDharkael** [comment]
They exist since GDNative 1.0 and I'm already using them.
They don't solve my problem as Write is useful tu prevent constant locking while accessing the PoolVector. When you create a Write, the locking happens only once and the data is unlocked when the Write goes out of scope.

**Comment by clayjohn** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!

**Note: This proposal may be irrelevant now with the new vector class in 4.0. You'll have to investigate before opening a proposal.**
